build_tools/cmake/iree_bytecode_module.cmake

```diff
@@ -56,22 +56,38 @@ function(iree_bytecode_module)
   if(DEFINED _RULE_TRANSLATE_TOOL)
     set(_TRANSLATE_TOOL ${_RULE_TRANSLATE_TOOL})
   else()
-    set(_TRANSLATE_TOOL "iree_tools_iree-translate")
+    set(_TRANSLATE_TOOL "iree-translate")
   endif()
 
-  # Resolve the executable binary path from the target name.
-  set(_TRANSLATE_TOOL_EXECUTABLE $<TARGET_FILE:${_TRANSLATE_TOOL}>)
+  if (CMAKE_CROSSCOMPILING)
+    iree_get_host_exectuable_path(${_TRANSLATE_TOOL} _TRANSLATE_TOOL_EXECUTABLE)
+  else()
+    # Resolve the executable binary path from the target name.
+    set(_TRANSLATE_TOOL_EXECUTABLE $<TARGET_FILE:${_TRANSLATE_TOOL}>)
+  endif()
 
   set(_ARGS "${_FLAGS}")
   list(APPEND _ARGS "${CMAKE_CURRENT_SOURCE_DIR}/${_RULE_SRC}")
   list(APPEND _ARGS "-o")
   list(APPEND _ARGS "${_RULE_NAME}.module")
 
-  add_custom_command(
-    OUTPUT "${_RULE_NAME}.module"
-    COMMAND ${_TRANSLATE_TOOL_EXECUTABLE} ${_ARGS}
-    DEPENDS ${_TRANSLATE_TOOL}
-  )
+  if (CMAKE_CROSSCOMPILING)
```

Comment:
we should try to remove this by having an alias for the executable to the short target name --- whether building for host or native 'iree-translate' should always be a valid dependency

Reply:
Yeah my general comment is that it would be nice if this didn't require configuration in basically every rule and we could instead make things work more generally on top of a layer that understands host vs target builds.

Reply:
@GMNGeoffrey: +1. Pushed more cross-compilation details inside various functions so that we can hide them from target declaration CMakeLists.txt. 

@benvanik: For here specifically, I think we can just depend on the file; that should work even if target is available.